src/detect.h

```diff
@@ -1040,6 +1040,9 @@ typedef struct DetectEngineCtx_ {
 
     /* Track rule requirements for reporting after loading rules. */
     SCDetectRequiresStatus *requirements;
+
+    /* number of signatures using filestore, limited as u16 */
+    uint16_t filestore_cnt;
```

Comment:
So, the limit for filestore sigs per Detection Engine context **and** per SGH is u16? But, there could be several SGHs in a Detection Engine context. How do we justify this?

Reply:
There is a limit put on Detection Engine Context, which implies a limit per SGH.

> How do we justify this?

Victor said in the last PR it was the way to go.
I guess the justification is that we do not expect people to use 65k different signatures with `filestore`.
If they do, they will complain that it does not work, and we will see what to do...

Reply:
Yeah, I think it's easier to enforce globally, as the sgh grouping is a bit of magic that is hard to understand for users.